devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -639,128 +692,177 @@ bool YarpRobotLoggerDevice::attachAll(const yarp::dev::PolyDriverList& poly)
 
     const unsigned dofs = joints.size();
     m_bufferManager.setDescriptionList(joints);
+    if(m_sendDataRT)
+    {
+        char* tmp = std::getenv("YARP_ROBOT_NAME");
+        std::string metadataName
+            = robotRtRootName + treeDelim + robotName;
+        m_vectorCollectionRTDataServer.populateMetadata(metadataName, {std::string(tmp)});
 
-    bool ok = true;
+        metadataName
+            = robotRtRootName + treeDelim + timestampsName;
+        m_vectorCollectionRTDataServer.populateMetadata(metadataName, {timestampsName});
+
+        std::string rtMetadataName = robotRtRootName + treeDelim
+                                    + robotDescriptionList;
+        m_vectorCollectionRTDataServer.populateMetadata(rtMetadataName, joints);
+    }
 
     // prepare the telemetry
     if (m_streamJointStates)
     {
-        ok = ok && m_bufferManager.addChannel({"joints_state::positions", {dofs, 1}, joints});
-        ok = ok && m_bufferManager.addChannel({"joints_state::velocities", {dofs, 1}, joints});
-        ok = ok && m_bufferManager.addChannel({"joints_state::accelerations", {dofs, 1}, joints});
-        ok = ok && m_bufferManager.addChannel({"joints_state::torques", {dofs, 1}, joints});
+        ok &= initMetadata(jointStatePositionsName, joints);
```

Comment:
I think this is a [bitwise and](https://stackoverflow.com/questions/33304407/c-meaning-and)
```suggestion
        ok = ok && initMetadata(jointStatePositionsName, joints);
```

Reply:
Fixed in [7c8d912](https://github.com/ami-iit/bipedal-locomotion-framework/commit/7c8d9125c7535a824cb27dc2cb7a8f03bebdff5a)